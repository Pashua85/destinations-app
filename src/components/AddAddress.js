import React from 'react';
import PlacesAutocomplete, {
  geocodeByAddress,
  getLatLng
} from 'react-places-autocomplete';

export default class AddAddress extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      address: ''
    };
  };

  handleFormSubmit = (event) => {
    event.preventDefault();
    
    if(this.state.address) {
      let newAddress = {};
      geocodeByAddress(this.state.address)
        .then(results => {
          newAddress.address = results[0].formatted_address;
          newAddress.place_id = results[0].place_id;
          newAddress.address_components = results[0].address_components;
          return getLatLng(results[0]);
        })
        .then(lanLng => {
          newAddress.lanLng = lanLng;
          console.log(newAddress);
          this.props.addAddress(newAddress);
          this.setState(() => ({ address: '' }))
        })
        .catch(error => {
          console.log('Error', error);
          this.setState(() => ({ address: ''}));
          this.props.handleError('Не удалось полусить данные о точке')
        })
    } else {
      this.props.handleError('Введите название точки маршрута');
    }
  };

  handleSelect = (address) => {
    this.setState(() => ({ address }))

    let newAddress = {};
    geocodeByAddress(address)
      .then(results => {
        newAddress.address = results[0].formatted_address;
        newAddress.place_id = results[0].place_id;
        newAddress.address_components = results[0].address_components;
        return getLatLng(results[0]);
      })
      .then(lanLng => {
        newAddress.lanLng = lanLng;
        console.log(newAddress);
        this.props.addAddress(newAddress);
        this.setState(() => ({ address: '' }))
      })
      .catch(error => {
        console.log('Error', error);
        this.setState(() => ({ address: ''}));
        this.props.handleError('Не удалось полусить данные о точке')
      })
  };

  handleChange = (address) => {
    this.setState(() => ({ address }));
  };

  onError = () => {};

  render() {
    const inputProps = {
      value: this.state.address,
      onChange: this.handleChange
    }

    return (
      <form onSubmit={this.handleFormSubmit} id="add-address-form">
        <PlacesAutocomplete 
          inputProps={inputProps} 
          onSelect={this.handleSelect}
          onEnterKeyDown={this.handleSelect}
          onError={this.onError}
        />
        <button type="submit">Добавить</button>
      </form>
    )
  }
};