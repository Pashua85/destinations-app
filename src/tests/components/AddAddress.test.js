import React from 'react';
import { shallow } from 'enzyme';
import AddAddress from '../../components/AddAddress';

test('should render AddAddress correctly', () => {
  const wrapper = shallow(<AddAddress />);
  expect(wrapper).toMatchSnapshot();
})