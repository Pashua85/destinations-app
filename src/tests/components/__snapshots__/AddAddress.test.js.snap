// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddAddress correctly 1`] = `
<form
  className="add-address-form"
  id="add-address-form"
  onSubmit={[Function]}
>
  <PlacesAutocomplete
    classNames={
      Object {
        "autocompleteContainer": "add-address__autocomplete-container",
        "input": "add-address__input",
        "root": "add-address__root",
      }
    }
    debounce={200}
    highlightFirstSuggestion={false}
    inputProps={
      Object {
        "onChange": [Function],
        "value": "",
      }
    }
    onEnterKeyDown={[Function]}
    onError={[Function]}
    onSelect={[Function]}
    options={Object {}}
    renderSuggestion={[Function]}
    shouldFetchSuggestions={[Function]}
    styles={Object {}}
  />
  <button
    className="add-address__btn"
    type="submit"
  >
    Add
  </button>
</form>
`;
